core/interfaces/i_procedure_map.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * The interface for the map of procedure models.
+ *
+ * @namespace Blockly.IProcedureMap
+ */
+import * as goog from '../../closure/goog/goog.js';
+goog.declareModuleId('Blockly.IProcedureMap');
+
+import {IProcedureModel} from './i_procedure_model.js';
+
+/**
+ * A special collection of procedure models.
+ */
+export interface IProcedureMap {
```

Comment:
Is there any reason not to just use `Map<string, IProcedureModel>` instead of this type?

That would seem to have various advantages: simplicity, principally, but also access to all of the excellent dynamic iteration facilities of `Map`, such as the feature that `for ([key, value] of map)` and `for value of Map.values())` correctly handling the case that items are deleted from the map during iteration.  (And if you really don't want that you can always `for (value of Array.from(…))`.)

Reply:
OK, I've gone and re-read the relevant section of the design doc.  I understand now why you have defined this, but I think a better approach would be to define a (concrete, general-purpose) `ObservableMap` subclass of `Map` that adds hooks to all the normal `Map` methods, and use instances of that—using variables/properties/parameters typed as `Map<…>` to hide the observability where appropriate.

Reply:
I don't think having a general-purpose map makes sense, because the events that this will be firing are specific to procedures. However I think subclassing `Map` is fine. Generally I prefer composition over inheritance because I think inheritance confuses subtyping with code reuse. But in this case... I'd be fine with it.

Reply:
> However I think subclassing Map is fine.

Are you planning to make the relevant changes in this PR (i.e., removing IProcedureMap or making it extend `Map`)?

Reply:
Done! Removed `IProcedureMap`